### PR TITLE
View plugin/command help in pager

### PR DESCRIPTION
**Pull request by slaykovsky**
ipa help code invokes pager if help lines length is more then
current terminal height.

https://pagure.io/freeipa/issue/7225

### CODE REVIEW

**Comment by abbra** [comment]
Will it work with output redirection? E.g. if I want `ipa help dns > dns-help.txt`, will that continue working as it is now, by producing a text file output?

**Comment by slaykovsky** [comment]
@abbra Yes, it works
```
[vagrant@master ~]$ ipa help topics | wc -l
45
[vagrant@master ~]$ ipa help topics > topics.txt
[vagrant@master ~]$ cat topics.txt | wc -l
45
```

File: ipalib/util.py [hunk 1]
```diff
@@ -1149,3 +1153,40 @@ def no_matching_interface_for_ip_address_warning(addr_list):
...
+    Args:
+        fd (int): file descriptor. Default: 1 (stdout)
+
+    Returns:
+        int: Terminal height
+    """
+    try:
+        return struct.unpack(
+            'hh', fcntl.ioctl(fd, termios.TIOCGWINSZ, b'1234'))[0]
+    except:
```
Comment (Tiboris) [inline ipalib/util.py]: Could you please add an 'Exception' after except? To 'specify' exception type (pylint err).

File: ipalib/cli.py [hunk 2]
```diff
@@ -680,10 +682,39 @@ def select_entry(self, entries, format, attrs, display_count=True):
         self.print_line('')
         return selection
 
+
 class help(frontend.Local):
     """
     Display help for a command or topic.
     """
+    class Writer:
```
Comment (Tiboris) [inline ipalib/cli.py]: Maybe adding 'object' inheritance could stop pylint from whining.

Comment (slaykovsky) [reply ipalib/cli.py]: It was not the case but thanks anyway :)

File: ipalib/util.py (hunk 1 above)
Comment (slaykovsky) [reply ipalib/util.py]: Thanks!

Comment (Tiboris) [reply ipalib/util.py]: No problem :)

File: ipalib/cli.py [hunk 3]
```diff
@@ -55,7 +55,9 @@
 from ipalib.parameters import File, Str, Enum, Any, Flag
 from ipalib.text import _
 from ipalib import api  # pylint: disable=unused-import
-from ipalib.util import check_client_configuration
+from ipalib.util import (
```
Comment (Tiboris) [inline ipalib/cli.py]: Could you please try to move it to line 40 before if statements?

**Comment by Tiboris** [comment]
Hi @slaykovsky.

now CI tests are failing for 'help' command  could you please have look into it?

**Comment by abbra** [comment]
master:

* caed210bc26c0220c9292c40b66177cda671a34f View plugin/command help in pager

**Pull request closed without merging**